middlewares/stripPrefix_test.go

```diff
@@ -0,0 +1,153 @@
+package middlewares
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStripPrefix(t *testing.T) {
+	handlerPath := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, r.URL.Path)
+	})
+
+	stripServers := make(map[string]*httptest.Server)
+	stripServers["/norules"] = httptest.NewServer(&StripPrefix{
+		Prefixes: []string{},
+		Handler:  handlerPath,
+	})
+
+	stripList := []string{
+		"/",
+		"/stat/",
+		"/stat",
+		"/nomatch",
+	}
+	for _, stripPath := range stripList {
+		stripServers[stripPath] = httptest.NewServer(&StripPrefix{
+			Prefixes: []string{stripPath},
+			Handler:  handlerPath,
+		})
```

Comment:
Initializing outside of the parallelized context could be problematic. For instance, if I execute your test like this

`go test -count 20 -cpu 1,2,4 -run ^TestStripPrefix$ ./middlewares`

I immediately run into `too many open files` problems on my Mac.

This might not be an issue now but could possibly become one in the future.